doc/user/content/ops/optimization.md

```diff
@@ -55,37 +59,146 @@ In general, your index key should exactly match the columns that are constrained
 - If `OR` is used and its arguments constrain completely disjoint sets of fields (e.g. `WHERE x = 5 OR y = 'aaa'`), try to rewrite your query using a `UNION` (or `UNION ALL`), where each argument of the `UNION` has one of the original `OR` arguments.
 
 ### `JOIN`
-Speed up a `JOIN` query by indexing the join keys:
 
-Clause                                      | Index                                                                       |
---------------------------------------------|-----------------------------------------------------------------------------|
-`FROM view V JOIN table T ON (V.id = T.id)` | `CREATE INDEX ON view (id);` <br /> `CREATE INDEX ON table (id);`           |
+In general, you can improve the performance of your joins by creating indexes on the columns being joined. This comes at the cost of additional memory usage. Fortunately, Materialize can reuse these indexes for different queries, which means **for multiple queries, indexes are a fixed upfront cost with memory savings for each new query.**
+
+Let's create a few tables to work through examples.
+
+```sql
+CREATE TABLE teachers (id INT, name TEXT);
+CREATE TABLE sections (id INT, teacher_id INT, course_id INT, schedule TEXT);
+CREATE TABLE courses (id INT, name TEXT);
+```
+
+#### Joining Two Collections
+
+Here is an example where we join a collection `teachers` to a collection `sections` to see the name of the teacher, schedule, and course ID for a specific section of a course.
+
+```sql
+SELECT
+    t.name,
+    s.schedule,
+    s.course_id
+FROM teachers t
+INNER JOIN sections s ON t.id = s.teacher_id;
+```
+
+We can optimize this query by creating an index for each column being joined.
+
+```sql
+CREATE INDEX pk_teachers ON teachers (id);
+CREATE INDEX sections_fk_teachers ON sections (teacher_id);
+```
+
```

Comment:
This section is not complete. If you run look at the plan without and then with the indexes it's the same - sans reading the data from persist. 

```
materialize=> EXPLAIN SELECT
    t.name,
    s.schedule,
    s.course_id
FROM teachers t
INNER JOIN sections s ON t.id = s.teacher_id;
               Optimized Plan
---------------------------------------------
 Explained Query:                           +
   Project (#1, #4, #3)                     +
     Join on=(#0 = #2) type=differential    +
       ArrangeBy keys=[[#0]]                +
         Filter (#0) IS NOT NULL            +
           Get materialize.public.teachers  +
       ArrangeBy keys=[[#0]]                +
         Project (#1..=#3)                  +
           Filter (#1) IS NOT NULL          +
             Get materialize.public.sections+
                                            +
 Source materialize.public.teachers         +
   filter=((#0) IS NOT NULL)                +
 Source materialize.public.sections         +
   filter=((#1) IS NOT NULL)                +

(1 row)

materialize=> CREATE INDEX pk_teachers ON teachers (id);
CREATE INDEX sections_fk_teachers ON sections (teacher_id);
CREATE INDEX
CREATE INDEX
materialize=> EXPLAIN SELECT
    t.name,
    s.schedule,
    s.course_id
FROM teachers t
INNER JOIN sections s ON t.id = s.teacher_id;
               Optimized Plan
---------------------------------------------
 Explained Query:                           +
   Project (#1, #5, #4)                     +
     Filter (#0) IS NOT NULL                +
       Join on=(#0 = #3) type=differential  +
         ArrangeBy keys=[[#0]]              +
           Get materialize.public.teachers  +
         ArrangeBy keys=[[#1]]              +
           Get materialize.public.sections  +
                                            +
 Used Indexes:                              +
   - materialize.public.pk_teachers         +
   - materialize.public.sections_fk_teachers+
```

That is because the index performs the first half of arranging the data. The benefit comes when you have a second view that also joins teachers or sections and can reuse the data in memory. We need to expand on that here.